external-crates/move/crates/move-compiler/tests/move_2024/typing/index_fun_invalid_visibility.move

```diff
@@ -0,0 +1,55 @@
+module 0x42::m {
+    // A
+
+    public struct A {}
+
+    #[syntax(index)]
+    fun index_a(a: &A): &A { a }
+
+    #[syntax(index)]
+    public fun index_a_mut(a: &mut A): &mut A { a }
+
+    public fun make_a(): &mut A { abort 0 }
+
+    // B
+
+    public struct B {}
+
+    #[syntax(index)]
+    public fun index_b(b: &B): &B { b }
+
+    #[syntax(index)]
+    fun index_b_mut(b: &mut B): &mut B { b }
+
+    public fun make_b(): &mut B { abort 0 }
+
+    // C
+
+    public struct C {}
+
+    #[syntax(index)]
+    fun index_c(c: &C): &C { c }
+
+    #[syntax(index)]
+    fun index_c_mut(c: &mut C): &mut C { c }
```

Comment:
Should we have a case for `public(package)` in this invalid test?

Reply:
Added as a new test